imaids/fieldsource.py

```diff
@@ -692,6 +692,34 @@ def calc_avg_period_length(
         nr_periods = int(len(dz)/2)
         return avg_period_length, nr_periods
 
+    def get_beff(self, polarization, hmax, x):
+        """Calculate effective field amplitude from model.
+        Args:
+            polarization (string): polarization of radiation. (['hp', 'vp', 'cp'])
```

Comment:
Here, I'd add a description of what hp, vp and cp mean.
An example is the block_type description in the docstiring of UndulatorShimming.\_\_init\_\_